src/change_stream.ts

```diff
@@ -969,15 +1026,16 @@ export class ChangeStream<
   }
 
   /** @internal */
-  private async _processErrorIteratorMode(changeStreamError: AnyError) {
+  private async _processErrorIteratorMode(changeStreamError: AnyError, cursorInitialized: boolean) {
     if (this[kClosed]) {
       // TODO(NODE-3485): Replace with MongoChangeStreamClosedError
       throw new MongoAPIError(CHANGESTREAM_CLOSED_ERROR);
     }
 
     if (
-      this.cursor.id == null ||
-      !isResumableError(changeStreamError, this.cursor.maxWireVersion)
+      !cursorInitialized ||
+      (!isResumableError(changeStreamError, this.cursor.maxWireVersion) &&
+        !(changeStreamError instanceof MongoOperationTimeoutError))
     ) {
```

Comment:
(optional) I struggle with lots of negated conditions - I find it easier to understand a single negation of a number of possive conditions.  Something like:

```suggestion
	const canResume = 
      cursorInitialized &&
      (isResumableError(changeStreamError) ||
        changeStreamError instanceof MongoOperationTimeoutError);
    if (!canResume) {
```

(`(!A || !B && !C) is equivalent to !(A && (B || C))`, I think)

Reply:
de morgan's law!